common-rendering/src/components/MetaData.tsx

```diff
@@ -0,0 +1,363 @@
+	);
+
+	return byline.split(regex);
+}; */
+
+/*export const getContributorTagsForToken = (
+	tags: TagType[],
+	token: string
+): TagType[] => {
+	return getContributorTags(tags).filter((t) => t.title === token);
+};*/
+
+/* export const getBylineComponentsFromTokens = (
+	tokens: string[],
+	tags: TagType[]
+): BylineComponent[] => {
+	const [bylineComponents] = tokens.reduce(
+		([bylines, remainingTags], token) => {
+			const [firstContributorTag] = getContributorTagsForToken(
+				remainingTags,
+				token
+			);
+			if (!firstContributorTag) {
+				return [bylines.concat(token), remainingTags];
+			}
+
+			// We've used this tag, so remove it from the list of possible tags.
+			// This ensures we don't use the same contributor tag when two identical
+			// names with different contributor tags are used.
+			const newRemainingTags = remainingTags.filter(
+				(tag) => !(tag.id === firstContributorTag.id)
+			);
+
+			return [
+				bylines.concat({ tag: firstContributorTag, token }),
+				newRemainingTags,
+			];
+		},
+		[[], tags] as [BylineComponent[], TagType[]]
+	);
+
+	return bylineComponents;
+}; */
+
+/* const BylineLink = ({ byline, tags }: Props) => {
+	const tokens = bylineAsTokens(byline, tags);
+
+	const bylineComponents = getBylineComponentsFromTokens(tokens, tags);
+
+	const renderedTokens = bylineComponents.map((bylineComponent) => {
+		if (typeof bylineComponent === "string") {
+			return bylineComponent;
+		}
+		return (
+			<ContributorLink
+				contributor={bylineComponent.token}
+				contributorTagId={bylineComponent.tag.id}
+				key={bylineComponent.tag.id}
+			/>
+		);
+	});
+
+	return <>{renderedTokens}</>;
+}; */
+
+/*
+	Foreach author in authors
+	if string contains name replace with authorlink then return <p> byline with links
+*/
+
+/* export const getContributorTags = (tags: TagType[]): TagType[] => {
+	return tags.filter((t) => t.type === "Contributor");
+}; */
+
+/* const getAuthorName = (tags: TagType[]) => {
+	const contributorTag = tags.find((tag) => tag.type === "Contributor");
+	return contributorTag && contributorTag.title;
+}; *
+
+type AuthorLinkDetails = {
+	id: string;
+	name: string;
+};
+
+const getAuthorNames = (tags: TagType[]) => {
+	const names: AuthorLinkDetails[] = [];
+	tags.find((tag) => {
+		if (tag.type === "Contributor") {
+			names.push({ name: tag.title, id: tag.id });
+		}
+	});
+};
+
+/*function Byline(byline: string, authors: Authors[]) {
+	if (byline.includes("now")) {
+		byline.replace("now", "(now)");
+	}
+	if (byline.includes("earlier")) {
+		byline.replace("earlier", "(earlier)");
+	}
+}*/
+
+const ContributorLink: React.FC<{
+	contributor: string;
+	contributorTagId: string;
+}> = ({ contributor, contributorTagId }) => (
+	<a
+		rel="author"
+		data-link-name="auto tag link"
+		href={`//www.theguardian.com/${contributorTagId}`}
+	>
+		{contributor}
+	</a>
+);
+
+// Shamelessly borrowed from Lucy! (This should be lifted out as it's very useful!)
```

Comment:
Can we add a `styles.ts` or similar to `common-rendering` containing this and other shared styling functions so we're not repeating at component level?

Reply:
We can do, this is mainly a draft until we decide what we are doing with DCR components going forward. There is a meeting later today (15/09/21)

Reply:
I guess it also depends on what the person needs. So I've modified Lucy's as I don't need the same data that she is using.